src/zoo.ts

```diff
@@ -0,0 +1,242 @@
+    this.issueTicket(ticketType);
+    this.increaseDaylyIncome(ticketType);
+  }
+  protected issueTicket(ticketType: TicketType): Ticket {
+    return (this._ticket = new Ticket(ticketType));
+  }
+  protected increaseDaylyIncome(ticketType: TicketType): number {
+    return (this._daylyIncome += ticketPrice[ticketType]);
+  }
+}
+
+class Ticket {
+  protected _ticketType!: TicketType;
+  protected _price!: number;
+  constructor(ticketType: TicketType) {
+    (this._price = ticketPrice[ticketType]), (this._ticketType = ticketType);
+  }
+}
+class Person {
+  private readonly _id!: number;
+  private _firstName!: string;
+  private _lastName!: string;
+  private _contacts?: string;
+  constructor(id: number, firstName: string, lastName: string, contacts: string) {
+    ((this._id = id), (this._firstName = firstName)), (this._lastName = lastName), (this._contacts = contacts);
+  }
+
+  public get firstName(): string {
+    return this._firstName;
+  }
+  public get lastName(): string {
+    return this._lastName;
+  }
+  public get id(): number {
+    return this._id;
+  }
+  public get contacts(): string | undefined {
+    return this._contacts;
+  }
+}
+const clientsList: Person[] = [];
+const currentVisitorsList: Person[] = [];
+
+class MarketingDepartment {
+  //  Використовує список клієнтів для розсилки новин про зоопарк і рекламні акції.
+  public sendAdvitisement(clientList: Person[], advertisement: string) {
+    clientList.forEach(client => {
+      console.log(`Sending ${advertisement} to ${client.firstName}`);
+    });
+  }
+}
+class Receipts {
+  //  "Виручка": Каса збирає дані про виручку за день. Ці дані передаються в Бухгалтерію.
+}
+class AccountDepartment {
+  //  "Виручка": Каса збирає дані про виручку за день. Ці дані передаються в Бухгалтерію.
+  // Розпоряджається бюджетом, включно з оплатою співробітників,
+  private _account: Account = new Account();
+  private _employee: Employer[] = [];
+  private _animals: Animal[] = [];
+
+  get employee(): Employer[] {
+    return this._employee;
+  }
+  get animals(): Animal[] {
+    return this._animals;
+  }
+  public closeKassa(kassa: Kassa): void {
+    const daylyIncome = kassa.closeDay();
+    this._account.getPayment(daylyIncome, TransactionTypeEnum.KASSA);
+  }
+  public paySalaries(): void {
+    this._employee.forEach(employee => this._account.makePayment(employee.salary, TransactionTypeEnum.SALARY));
+    // this.balance -= employee.reduce((acc, employee)=>acc += employee.salary,0)
+  }
+  //  "Адміністрація": Відповідає за управління співробітниками і тваринами. Може додавати і видаляти співробітників і тварин.
+  public addListManager(item: Employer | Animal): void {
+    const list: (Employer | Animal)[] = isAnimal(item) ? this._animals : this._employee;
+    if (!list.includes(item)) {
+      list.push(item);
+    } else {
+      console.log('This item is already in the list');
+    }
+  }
+  public removeListManager(item: Employer | Animal): void {
```

Comment:
Just a quick heads-up - using simple and clear method names like 'addManager' and 'removeManager' is usually best. It helps keep things straightforward and easy to understand.